enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,686 @@
+enhancement handled `spec.loadBalancer.platform.aws.lbType`.
+
+The OpenShift Installer will be updated to support the new `platform.aws.subnetsConfig` field, and the
+Ingress Operator will be updated to apply the `Subnets` field in the IngressConfig as the default subnets
+for new IngressControllers.
+
+### Implementation Details/Notes/Constraints
+
+As mentioned in [Proposal](#proposal), the install-config and the IngressConfig will be updated
+to support propagating default subnets to all new IngressControllers, including the default
+IngressController. The Ingress Operator will also need to be updated to use the new IngressConfig
+`Subnet` field.
+
+#### Installer Updates
+
+The `platform.aws.subnets` field in the install-config will be deprecated and replaced with a new
+field, `platform.aws.subnetsConfig`. Much like `Subnets`, the new `SubnetsConfig` field indicates
+a list of subnets in a pre-existing VPC, but also provide the role that the subnet will fulfill in the
+cluster. Additionally, since it is a struct and not a `[]string`, it provide the ability to be
+expanded with additional subnet-related fields in the future.
+
+```go
+// Platform stores all the global configuration that all machinesets
+// use.
+// +kubebuilder:validation:XValidation:rule=`!(has(self.subnets) && has(self.subnetsConfig))`,message="cannot use both subnets and subnetsConfig"
+type Platform struct {
+	[...]
+    // Subnets specifies existing subnets (by ID) where cluster
+    // resources will be created.  Leave unset to have the installer
+    // create subnets in a new VPC on your behalf.
+	//
+    // Deprecated: Use SubnetsConfig
+    //
+    // +optional
+    Subnets []string `json:"subnets,omitempty"`
+
+	// SubnetsConfig specifies the subnet configuration for
+	// a cluster by specifying a list of subnet ids with their
+	// designated role. At least one Cluster role subnet must be
+	// specified. If no Ingress role subnets are specified, the
+	// IngressController's Load Balancer will automatically discover
+	// its subnets based on the kubernetes.io/cluster/<cluster-id> tag,
+	// whether it's public or private, and the availability zone.
+	// Leave this field unset to have the installer create subnets
+	// in a new VPC on your behalf.
+	//
+	// +optional
+    // +listType=atomic
+    // +kubebuilder:validation:XValidation:rule=`self.all(x, self.exists_one(y, x.id == y.id))`,message="subnetsConfig cannot contain duplicate IDs"
+    // +kubebuilder:validation:XValidation:rule=`self.exists(x, x.roles.exists(r, r == 'Cluster'))`,message="subnetsConfig must contain at least 1 subnet with the Cluster role"
+    // +kubebuilder:validation:XValidation:rule=`self.filter(x, x.roles.exists(r, r == 'Ingress')).size() <= 10`,message="subnetsConfig must contain less than 10 subnets with the Ingress role"
+	// +openshift:enable:FeatureGate=IngressControllerLBSubnetsAWS
```

Comment:
Can you refer to a feature gate in the installer?  How can the feature gate be enabled if there is no cluster?

Reply:
Whoops, no you can't. The installer has the concept of feature sets, but are enforced differently. For example, the [NatGatewayOutboundType in Azure](https://github.com/openshift/installer/blob/364a47391562d35662e64503c5e7052d06da6aec/pkg/types/azure/validation/platform.go#L103) is only supported in tech preview. I am just learning how the installer works, so I expect I may get some feedback from them.

Fixed and updated with details on tech preview vs. GA.